src/main/kotlin/cacophony/controlflow/generation/CFGGenerator.kt

```diff
@@ -46,7 +37,14 @@ internal class CFGGenerator(
                 }
             }
 
-        val returnVertex = cfg.addFinalVertex(CFGNode.Return)
+        val resultSize =
+            when (function.returnType) {
+                is BaseType.Basic -> 1
+                is BaseType.Structural -> function.returnType.fields.size
```

Comment:
that doesn't cover nested structures, we need to calculate size of return type after flattening